docs/howto/how-to-contribute.md

```diff
@@ -0,0 +1,112 @@
+# Contributing
+
+Thank you for your interest in contributing to the CredSweeper tool!
+
+The document covers the process for contributing to the CredSweeper code and documentation. Contributions may be as simple as typo corrections or as complex as new features.
+
+1.  [Process for contributing](#process-for-contributing)
+    1. [Repository structure](#repository-structure)
+    1. [File Name](#file-name)
+    1. [Self Test & Verification](#self-test-and-verification)
+1.  [How to PR](#how-to-pr)
+1.  [DOs and DON'Ts](#dos-and-donts)
+
+
+## Process for contributing
+
+You need a basic understanding of [Git and GitHub.com](https://guides.github.com/activities/hello-world/).
+
+**Step 1:** You can skip this step for small changes such as typo corrections. Open an [new issue](https://github.com/Samsung/CredSweeper/issues/new) describing what you want to do, such as change an existing code, functionality or create a new one.
+
+You can also look at our [open issues](https://github.com/Samsung/CredSweeper/issues) list and volunteer to work on the ones you're interested in.
+
+**Step 2:** Fork the `/Samsung/CredSweeper` repo and create a branch for your changes.
+
+For small changes, you can use GitHub's web interface. Simply click the **Edit the file in your fork of this project** on the file you'd like to change.
+GitHub creates the new branch for you when you submit the changes.
+
+**Step 3:** Make the changes on this new branch.
+
+Be sure to follow the proper Python syntax. For more information, see the [style guide](./.style.yapf).
```

Comment:
`.style.yapf` is not a good user manual, but a yapf config file can can be applied for automated reformatting

Yapf can be installed as a pre-commit hook with
```bash
pip install pre-commit
pre-commit install
```

I think it might be useful to add this to step to `How to PR`